src/snowflake/connector/connection.py

```diff
@@ -1094,3 +1104,101 @@ def __exit__(self, exc_type, exc_val, exc_tb):
         else:
             self.rollback()
         self.close()
+
+    def _get_query_status(self, sf_qid: str) -> Tuple[QueryStatus, Dict[str, Any]]:
+        """Retrieves the status of query with sf_qid and returns it with the raw response.
+
+        This is the underlying function used by the public get_status functions.
+
+        Args:
+            sf_qid: Snowflake query id of interest.
+        """
+        logger.debug('get_query_status sf_qid=\'{}\''.format(sf_qid))
+
+        status = 'NO_DATA'
+        status_resp = self.rest.request('/monitoring/queries/' + quote(sf_qid),
+                                        method='get',
+                                        client='rest')
+        queries = status_resp['data']['queries']
+        if len(queries) > 0:
+            status = queries[0]['status']
+        status_ret = QueryStatus[status]
+        # If query was started by us and it has finished let's cache this info
+        if sf_qid in self._async_sfqids and not self.is_still_running(status_ret):
+            self._async_sfqids.remove(sf_qid)
+            self._done_async_sfqids.add(sf_qid)
+        return status_ret, status_resp
+
+    def get_query_status(self, sf_qid: str) -> QueryStatus:
+        """Retrieves the status of query with sf_qid.
+
+        Query status is returned as a QueryStatus.
+
+        Args:
+            sf_qid: Snowflake query id of interest.
+        """
+        status, status_resp = self._get_query_status(sf_qid)
+        return status
+
+    def get_query_status_throw_if_error(self, sf_qid: str) -> QueryStatus:
+        """Retrieves the status of query with sf_qid as a QueryStatus and raises an exception if the query terminated with an error.
+
+        Query status is returned as a QueryStatus.
+
+        Args:
+            sf_qid: Snowflake query id of interest.
+        """
+        status, status_resp = self._get_query_status(sf_qid)
+        queries = status_resp['data']['queries']
+        if self.is_an_error(status):
+            if sf_qid in self._async_sfqids:
+                self._async_sfqids.remove(sf_qid)
+            message = status_resp.get('message')
+            if message is None:
+                message = ''
+            code = status_resp.get('code')
+            if code is None:
+                code = -1
+            sql_state = None
+            if 'data' in status_resp:
+                message += queries[0].get('errorMessage', '') if len(queries) > 0 else ''
+                sql_state = status_resp['data'].get('sqlState')
+            Error.errorhandler_wrapper(self, None,
```

Comment:
@sfc-gh-mkeller It seems the message is meant to be exposed to the customer correct? I was just wondering if there's a possibility that the 'message' field has something that we don't want to show / log. Like some credentials / signatures?

Reply:
I have made sure to raise Exceptions the same way as it is done when someone executes a synchronous query. So I hope that the way this is raising exceptions is appropriate 😅 

https://github.com/snowflakedb/snowflake-connector-python/blob/f4fe2277b82d98f2122478d5df712c59420ea0bd/src/snowflake/connector/cursor.py#L590